hlxsites/vg-macktrucks-com
# Add custom decorateIcons script for helix v5 #704

Fix #704

Test URLs:
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/
- After: https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/

Changes:
- Move `decorateIcons` script to common.js
- Refactor usage of icons in several blocks; remove inlined SVG code and leverage SVG sprite
- Refactor usage of logo in footer/header; add animation as per design
- Remove slow scroll to top functionality; [apply code of Volvo Trucks](https://github.com/Netcentric/vg-volvotrucks-us-rd/commit/5ac6f9b18669b0de4c612f085da6f7d802a51918)
- Apply `:focus-visible` to links and form in the footer
- Replace hardcoded content with text defined in placeholders.json for these blocks:
  - header
  - footer
  - modal
  - v2-feature-carousel
  - v2-images-grid
  - v2-truck-lineup

The following blocks are changed:
| Block | Test URL | Comment |
|--------|--------|--------|
| [header] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/) | Hover over logo; check only on desktop |
| [footer] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/) | Check logo icon, focus states, functionality of scroll to top button |
| [modal] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/) | Open modal in link in hero |
| [v2-accordion] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-accordion&index=0) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-accordion&index=0) | Simple regression test of icons |
| [v2-accordion-column] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/trucks/segment-landing) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/trucks/segment-landing) | Simple regression test of icons |
| [v2-feature-carousel] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-feature-carousel&index=0) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-feature-carousel&index=0) | Simple regression test of icons |
| [v2-hotspots] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-hotspots&index=0) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-hotspots&index=0) | Simple regression test of icons |
| [v2-images-grid] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-images-grid&index=0) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/v2-images-grid&index=0) | Arrows when modal of block is open; check only on desktop |
| [v2-inpage-navigation] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/pdp/lr) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/pdp/lr) | Dropdown icon; check only on mobile |
| [v2-product-listing] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/trucks/all-trucks) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/trucks/all-trucks) | Dropdown icon; check only on mobile |
| [v2-truck-lineup] | [Before](https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/homepage) [After](https://704-custom-decorateicons--vg-macktrucks-com--hlxsites.hlx.page/drafts/syb/page/homepage) | Arrows on hover; check only on desktop |
